tf2/include/tf2/impl/convert.h

```diff
@@ -26,53 +26,550 @@
+  // void toMsg(const Datatype&, Message&);
+  // void fromMsg(const Message&, Datatype&);
+};
+
+/**
+ * \brief Mapping of unstamped Messages for stamped Messages
+ *
+ * This struct contains utility methods to access the data member of a stamped ROS message
+ * and an alias (named \c UntampedType ) of the unstamped message type.
+ * It is needed for the conversion of stamped datatypes,
+ * so that only the conversions of unstamped datatypes has do be implemented.
+ * For example, a \c geometry_msgs::Vector3Stamped has two members,
+ * the \c header (which contains a timestamp and a frame ID) and the \c vector itself.
+ * For this class, the specialization should look like
+ * \code
+ * template<>
+ * struct StampedMessageTraits<geometry_msgs::Vector3Stamped>
+ * {
+ *  using UntampedType = geometry_msgs::Vector3;
+ *  static geometry_msgs::Vector3& accessMessage(geometry_msgs::Vector3Stamped& vs)
+ *  {
+ *     return vs.vector;
+ *  }
+ *  static geometry_msgs::Vector3 getMessage(const geometry_msgs::Vector3Stamped& vs)
+ *  {
+ *     return vs.vector;
+ *  }
+ * };
+ * \endcode
+ * The both almost identical methods are required to keep const-correctness.
+ *
+ * If the message is a stamped message with a covariance member,
+ * accessMessage() should return the underlying message (e.g. Pose)
+ * and accessCovariance() should return the covariance accordingly.
+ *
+ * \tparam StampedMessage The datatype of the ros message
+ */
+template<class StampedMessage>
+struct StampedMessageTraits
+{
+  static_assert(
+    alwaysFalse<StampedMessage>,
+    "No traits for this stamped message type available, "
+    "please check your header file includes!");
+  // using UntampedType = ...;
+  // static UntampedType& accessMessage(StampedMsg &);
+  // static UntampedType getMessage(StampedMsg const&);
+  // static CovarianceType & accessCovariance(MsgWithCovarianceStamped &);
+  // static CovarianceType getCovariance(MsgWithCovarianceStamped const&);
+};
+
+/**
+ * \brief Mapping of stamped Messages for unstamped Messages
+ *
+ * This struct is needed for the deduction of the return type of
+ * tf2::convert() for tf2::Stamped\<\> datatypes.
+ * Its specializations should contain an alias (named \c StampedType )
+ * of the stamped type.
+ * Example:
+ * \code
+ * template<>
+ * struct UnstampedMessageTraits<geometry_msgs::Vector3>
+ * {
+ *    using StampedType = geometry_msgs::Vector3Stamped;
+ * };
+ * \endcode
+ *
+ * If messages with covariance are also available,
+ * an alias with the name \c StampedTypeWithCovariance
+ * to the accoring message type should be declared.
+ *
+ * \tparam UnstampedMessage Type of the ROS message which is not stamped
+ */
+template<class UnstampedMessage>
+struct UnstampedMessageTraits
+{
+  static_assert(
+    alwaysFalse<UnstampedMessage>,
+    "No traits for this message type available, "
+    "please check your header file includes!");
+  // using StampedType = ...;
+  // using StampedTypeWithCovariance = ...;
```

Comment:
remove ?
if you don't want to remove these comments at least add a comment explaining the meaning. (some for all)